consensus/src/ledger/merkle.rs

```diff
@@ -21,13 +21,14 @@ use anyhow::*;
 use indexmap::IndexSet;
 use snarkos_storage::Digest;
 use snarkvm_algorithms::MerkleParameters;
+use tokio::task;
 
 use super::indexed_merkle_tree::IndexedMerkleTree;
 
 pub struct MerkleLedger<P: MerkleParameters> {
     ledger_digests: IndexSet<Digest>,
-    commitments: IndexedMerkleTree<P>,
-    serial_numbers: IndexedMerkleTree<P>,
+    commitments: Option<IndexedMerkleTree<P>>,
```

Comment:
can we decompose and recompose the merkle ledger rather than taking ownership like this? it's messy.

Reply:
I wanted to do that, but we can't take ownership of `Ledger`, as it would make it non-object-safe, and remove the possibility of using it as a trait object.

Reply:
and I think that we can't do this with `MerkleLedger`, as we need the `async` access at a higher layer

Reply:
I will double-check, though :+1:

Reply:
rather than making `extend` async, just keep it sync, have the caller (or a default impl in the trait definition) manage the async end and calling the check for should spawn a blocking task. keep this struct pure to its function -- and we can pass the merkle ledger entirely or by reference without issue.

Reply:
there is also some tricks with swapping a `&mut self` with a default impl and back again if you really need to.

Reply:
I considered `mem::take`, but the default impl wasn't as trivial as I like; however, it might be negligible; I'll try to make this change simpler